lib/segment/src/index/hnsw_index/graph_layers.rs

```diff
@@ -449,7 +469,7 @@ mod tests {
         assert_eq!(main_entry.level, num_levels);
 
         let total_links_0 = (0..num_vectors)
-            .map(|i| graph_layers.links.links(i as PointOffsetType, 0).len())
+            .map(|i| graph_layers.links.links(i as PointOffsetType, 0).count())
```

Comment:
Nit: if we can return an exact size iterator here, we can simply do [len()](https://doc.rust-lang.org/std/iter/trait.ExactSizeIterator.html#method.len). 

That may be a bit nicer because it'll prevent evaluating each entry. If we add decompression to the mix it'll prevent decompressing entries on getting the length.

Count does not have this _optimization_ and will evaluate all entries

Wdyt? Would that be worth it?